bundles/admin/admin-layereditor/instance.js

```diff
@@ -362,28 +364,26 @@ Oskari.clazz.defineES('Oskari.admin.admin-layereditor.instance',
                 scroll: false
             });
             this.themeFlyout.setSaveAction((value, id) => {
-                const httpMethod = id ? 'POST' : 'PUT';
-                const getpayload = () => {
-                    if (id) {
-                        return parentId ? { locales: value, id: id, parentId: parentId } : { locales: value, id: id };
-                    } else {
-                        return { locales: value };
-                    }
-                };
+                const httpMethod = newSubgroup ? 'PUT' : id ? 'POST' : 'PUT';
+
+                const payload = newSubgroup ? { locales: value, parentId: id } : id ? { locales: value, id: id, parentId: parentId } : { locales: value };
```

Comment:
Simplify these lines. It's easier to maintain if you don't try one-lining this.

Reply:
This has been fixed.